5.0/fr/0x10-V1-Encoding-and-Sanitization.md

```diff
@@ -0,0 +1,103 @@
+
+La validation des entrées constitue un mécanisme de défense en profondeur contre les contenus inattendus ou dangereux. Cependant, son objectif principal étant de garantir que le contenu entrant répond aux attentes fonctionnelles et métier, les exigences y afférentes sont décrites dans le chapitre « Validation et logique métier ».
+
+## V1.1 Architecture d'encodage et de nettoyage
+
+Les sections ci-dessous présentent les exigences spécifiques à la syntaxe ou à l'interpréteur pour traiter en toute sécurité les contenus non sécurisés afin d'éviter les failles de sécurité. Ces exigences précisent l'ordre et le lieu de traitement. Elles visent également à garantir que les données stockées conservent leur état d'origine et ne sont pas stockées sous une forme encodée ou échappée (par exemple, encodage HTML), afin d'éviter les problèmes de double encodage.
+
+| # | Description | Niveau |
+| :---: | :--- | :---: |
+| **1.1.1** | Vérifiez que l'entrée est décodée ou non échappée dans une forme canonique une seule fois, elle n'est décodée que lorsque des données codées sous cette forme sont attendues, et que cela est fait avant de traiter davantage l'entrée, par exemple, elle n'est pas effectuée après la validation ou la désinfection de l'entrée. | 2 |
+| **1.1.2** | Vérifiez que l'application effectue l'encodage et l'échappement de sortie soit comme étape finale avant d'être utilisée par l'interpréteur pour lequel elle est destinée, soit par l'interpréteur lui-même. | 2 |
+
+## V1.2 Prévention des injections
+
+L'encodage ou l'échappement de sortie, effectué à proximité ou à proximité d'un contexte potentiellement dangereux, est essentiel à la sécurité de toute application. Généralement, l'encodage et l'échappement de sortie ne sont pas conservés, mais servent à sécuriser la sortie pour une utilisation immédiate dans l'interpréteur approprié. Une tentative d'exécution trop précoce peut entraîner une altération du contenu ou rendre l'encodage ou l'échappement inefficaces.
+
+Dans de nombreux cas, les bibliothèques logicielles incluent des fonctions sûres ou plus sûres qui exécutent cela automatiquement, même s'il est nécessaire de s'assurer qu'elles sont correctes pour le contexte actuel.
+
+| # | Description | Niveau |
+| :---: | :--- | :---: |
+| **1.2.1** | Vérifiez que l'encodage de sortie d'une réponse HTTP, d'un document HTML ou d'un document XML est pertinent pour le contexte requis, comme l'encodage des caractères pertinents pour les éléments HTML, les attributs HTML, les commentaires HTML, les CSS ou les champs d'en-tête HTTP, pour éviter de modifier la structure du message ou du document. | 1 |
+| **1.2.2** | Lors de la création dynamique d'URL, vérifiez que les données non fiables sont cenodées en fonction de leur contexte (par exemple, encodage d'URL ou encodage base64url pour les paramètres de requête ou de chemin). Assurez-vous que seuls les protocoles d'URL sûrs sont autorisés (par exemple, interdire javascript: ou data:). | 1 |
```

Comment:
I would keep "Verify that" at the beginning of the sentence as far as possible: "Vérifiez que, lors de la création dynamique d'URL, les données non fiables …"